include/fiction/networks/views/bfs_topo_view.hpp

```diff
@@ -0,0 +1,295 @@
+ * Only reachable nodes are included in `foreach_node` and `foreach_gate`, which may exclude some network nodes.
+ *
+ * Required network functions:
+ * - `get_constant`
+ * - `foreach_pi`
+ * - `foreach_fanin`
+ * - `incr_trav_id`
+ * - `set_visited`
+ * - `trav_id`
+ * - `visited`
+ *
+ * @tparam Ntk Logic network type.
+ * @tparam sorted Internal tag to indicate if the network is already topologically sorted (default: false).
+ */
+template <class Ntk, bool sorted = mockturtle::is_topologically_sorted_v<Ntk>>
+class bfs_topo_view
+{};
+
+/**
+ * Specialization for non-topologically sorted networks.
+ *
+ * @tparam Ntk Logic network type.
+ */
+template <typename Ntk>
+class bfs_topo_view<Ntk, false> : public mockturtle::immutable_view<Ntk>
+{
+  public:
+    using storage                                 = typename Ntk::storage;
+    using node                                    = typename Ntk::node;
+    using signal                                  = typename Ntk::signal;
+    static constexpr bool is_topologically_sorted = true;
+
+    /**
+     * Constructs a breadth-first topological view from a given network.
+     *
+     * @param ntk Logic network to be wrapped with a topological view.
+     */
+    explicit bfs_topo_view(Ntk const& ntk) : mockturtle::immutable_view<Ntk>(ntk), fanout_ntk(ntk)
+    {
+        static_assert(mockturtle::is_network_type_v<Ntk>, "Ntk is not a network type");
+        static_assert(mockturtle::has_size_v<Ntk>, "Ntk does not implement the size method");
+        static_assert(mockturtle::has_get_constant_v<Ntk>, "Ntk does not implement the get_constant method");
+        static_assert(mockturtle::has_foreach_pi_v<Ntk>, "Ntk does not implement the foreach_pi method");
+        static_assert(mockturtle::has_foreach_fanin_v<Ntk>, "Ntk does not implement the foreach_fanin method");
+        static_assert(mockturtle::has_incr_trav_id_v<Ntk>, "Ntk does not implement the incr_trav_id method");
+        static_assert(mockturtle::has_set_visited_v<Ntk>, "Ntk does not implement the set_visited method");
+        static_assert(mockturtle::has_trav_id_v<Ntk>, "Ntk does not implement the trav_id method");
+        static_assert(mockturtle::has_visited_v<Ntk>, "Ntk does not implement the visited method");
+
+        update_bfs_topo();
+    }
+
+    /**
+     * Returns the number of nodes in the topological view.
+     *
+     * @return Total number of visited nodes in the topological order.
+     */
+    auto size() const
+    {
+        return static_cast<uint32_t>(topo_order.size());
+    }
+
+    /**
+     * Returns the number of gates in the topological view.
+     *
+     * @return Number of gates excluding constants and primary inputs.
+     */
+    auto num_gates() const
+    {
+        uint32_t const offset = 1u + this->num_pis() +
+                                (this->get_node(this->get_constant(true)) != this->get_node(this->get_constant(false)));
```

Comment:
We use west-`const` in *fiction*.

```suggestion
        const uint32_t offset = 1u + this->num_pis() +
                                (this->get_node(this->get_constant(true)) != this->get_node(this->get_constant(false)));
```

Reply:
Please check everywhere.